common/src/gaussian.cpp

```diff
@@ -183,37 +182,39 @@ pcl::GaussianKernel::convolveCols (const pcl::PointCloud<float>& input,
   assert (kernel.size () % 2 == 1);
   std::size_t kernel_width = kernel.size () -1;
   std::size_t radius = kernel.size () / 2;
-  const pcl::PointCloud<float>* input_;
+  std::unique_ptr<const pcl::PointCloud<float>> copied_input;
+  const pcl::PointCloud<float>* unaliased_input;
   if (&input != &output)
   {
     if (output.height < input.height || output.width < input.width)
     {
       output.width = input.width;
       output.height = input.height;
-      output.resize (input.width * input.height);
+      output.points.resize (input.height * input.width);
     }
-    input_ = &input;
+    unaliased_input = &input;
   }
   else
-    input_ = new pcl::PointCloud<float> (input);
+  {
+    copied_input = std::make_unique<pcl::PointCloud<float>>(input);
+    unaliased_input = copied_input.get();
+  }
 
   std::size_t j;
-  for (std::size_t i = 0; i < input_->width; i++)
+  for (std::size_t i = 0; i < unaliased_input->width; i++)
   {
     for (j = 0 ; j < radius ; j++)
       output (i,j) = 0;
 
-    for ( ; j < input_->height - radius ; j++)  {
+    for ( ; j < unaliased_input->height - radius ; j++)  {
       output (i,j) = 0;
       for (int k = static_cast<int>(kernel_width), l = static_cast<int>(j - radius) ; k >= 0 ; k--, l++)
       {
-        output (i,j) += (*input_) (i,l) * kernel[k];
+        output (i,j) += (*unaliased_input) (i,l) * kernel[k];
       }
     }
 
-    for ( ; j < input_->height ; j++)
+    for ( ; j < unaliased_input->height ; j++)
```

Comment:
1) What is the adjusted code doing?
2) Isn't it the same code as 50 lines above? Could you do code deduplication?

Reply:
1. no need of manually managing the local memory by conditionally creating a new memory block to unalias input and output pointers
2. Almost. And yes. But maybe not in this PR